platform/osgi-platform/src/main/java/org/motechproject/server/osgi/status/PlatformStatusManagerImpl.java

```diff
@@ -34,14 +55,22 @@ public void bundleChanged(BundleEvent bundleEvent) {
         if (bundleEvent.getType() == BundleEvent.STARTED) {
             LOGGER.debug("Bundle {} started", symbolicName);
             platformStatus.addOSGiStartedBundle(symbolicName);
+            checkBundleOsgiStarted(bundleEvent.getBundle());
         } else if (bundleEvent.getType() == BundleEvent.STOPPED) {
             LOGGER.trace("Bundle {} stopped", symbolicName);
             platformStatus.removeOSGiStartedBundle(symbolicName);
+            osgiBundles.remove(bundleEvent.getBundle());
```

Comment:
why are we just removing and not calling the check method here? If this happens for the last bundle in line, the event will not be fired.